### DISCUSSION TITLE
How to set properly the working directory in Spring projects?

### DISCUSSION THREAD

**Discussion by EgorkaKulikov** [UnitTestBot/UTBotJava#2348] [Q&A]
Having the proper value of working directory during Spring project analysis is very important. For example, `file:` annotations processing is aliased on the working directory.

Currently `Project.baseDir` property from Intellij psi elements hierarchy is used as working directory. However, documentation says that it is not reliable. 

- What should we do in the case of mismatch? Should we inform user that he can not only increase timeout, but also change the working directory value?
- How can we give a chance to change the value? With settings tab? Or we should not do it at all because harmful effects may be stronger?

**Comment by alisevych** [comment]
Additional questions:
- Can we get the value from Maven/Gradle? 
- It is about correct `user.dir`, correct? https://docs.oracle.com/javase/8/docs/api/index.html?java/io/File.html
